Documentation/Input/CreateDataProvider.md

```diff
@@ -0,0 +1,233 @@
+    fileName = "MixedRealityInputSimulationProfile",
+    order = (int)CreateProfileMenuItemIndices.InputSimulation)]
+public class MixedRealityInputSimulationProfile : BaseMixedRealityProfile
+{ }
+```
+
+The `CreateAssetMenu` attribute can be applied to the profile class to enable customers to create a profile instance using the 
+**Create > Assets > Mixed Reality Toolkit > Profiles** menu.
+
+### Implement the inspector
+
+Profile inspectors are the user interface for configuring and viewing profile contents. Each profile inspector should extend the
+[`BaseMixedRealityToolkitConfigurationProfileInspector]() class.
+
+``` c#
+[CustomEditor(typeof(MixedRealityInputSimulationProfile))]
+public class SpatialObjectMeshObserverProfileInspector : BaseMixedRealityToolkitConfigurationProfileInspector
+{ }
+```
+
+The `CustomEditor` attribute informs Unity the type of asset to which the inspector applies.
+
+## Create assembly definition(s)
+
+The Mixed Reality Toolkit uses assembly definition ([.asmdef](https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html)) files to specify dependencies
+between components as well as to assist Unity in reducing compilation time.
+
+It is recommended that assembly definition files are created for all data providers and their editor components.
+
+Using the [folder structure](#recommended-folder-structure) in the earlier example, there would be two .asmdef files for the ContosoInput data provider.
+
+The first assembly definition is for the data provider. For this example, it will be called ContosoInput and will be located in the example's ContosoInput folder.
+This assembly definition must specify a dependency on Microsoft.MixedReality.Toolkit and any other assemblies upon which it depends.
+
+The ContosoInputEditor assembly definition will specify the profile inspector and any editor specific code. This file must be located in the root folder of the editor code. In this example,
+the file will be located in the ContosoInput\Editor folder. This assembly definition will contain a reference to the ContosoInput assembly as well as:
+
+- Microsoft.MixedReality.Toolkit
+- Microsoft.MixedReality.Toolkit.Editor.Inspectors
+- Microsoft.MixedReality.Toolkit.Editor.Utilities
+
+## Register the data provider
+
+Once created, the data provider can be registered with the input system be used in the application.
+
+![Registered input system data providers](../Images/Input/RegisteredServiceProviders.png)
+
+## Packaging and distribution
+
+Data providers that are distributed as third party components have the specific details of packaging and distribution left to the preference of the developer. Likely, the most
+common solution will be to generate a .unitypackage and distribute via the Unity Asset Store.
+
+If a data provider is submitted and accepted as a part of the Microsoft Mixed Reality Toolkit package, the Microsoft MRTK team will package and distribute it as part of the MRTK
+offerings.
```

Comment:
This is a super nit, but there are quite a few spots in these docs where lines have a newline in the middle of them (like here, with one word alone on a new line). Though it doesn't have an effect on the rendered docs, it reads a little weird in the raw markdown.

Reply:
i believe it is related to my trying to limit line length... i am more than happy to address

Reply:
Ah that makes sense!